### DISCUSSION TITLE
macOS Sonoma Beta Game Mode not turning on for osu

### DISCUSSION THREAD

**Discussion by sackboy19** [ppy/osu#24557] [Q&A] [answered]
Hello, so in the newest version of macOS, Sonoma, there is a feature called Game Mode which should automatically activate when you open a game, and it will give the game top CPU and GPU priority and also improves latency with wireless devices, and doubles bluetooth sample rate (giving airpods and stuff less latency).

My question is, why does this mode not turn on for osu? It works in basically all my other games like minecraft and all steam games.

Game Mode on for Minecraft:
<img width="1440" alt="image" src="https://github.com/ppy/osu/assets/83356017/77e3654e-36d7-4759-9fd4-87279d7f9eaf">

Game Mode not appearing on osu:
<img width="1440" alt="image" src="https://github.com/ppy/osu/assets/83356017/47e4c2e7-83c9-4458-b104-52ec1f16f47d">

I'm just asking to see if its an issue with Sonoma itself and needs to be reported to apple, or if it has to due with the way osu lazer handles its application.

I tried on all window modes (fullscreen, borderless, windowed) and none of them seem to turn on game mode.

If anyone could please help me out because whenever I try to play osu with my airpods the audio is very delayed and I would benefit heavily from the doubled sample rate. Thank you!

**Comment by peppy** [comment]
no idea

**Comment by Jinkku** [comment]
Just a guess. It might be because osu! is not whitelisted for Game Mode maybe. Sonoma is still in beta so it might not be detected correctly

**Reply by sackboy19** [reply]
@pxkidoescoding I think it has to do with how osu is built, I made an xcode app and set the App Category to "Games - Music Games", and when I ran my app and went into fullscreen, it turned on game mode.

**Comment by sackboy19** [comment, marked as answer]
Ok, I ended up getting it working by building my own version of the game and setting the LSApplicationCategoryType in the Info.plist to "public.app-category.music-games"

<img width="795" alt="image" src="https://github.com/ppy/osu/assets/83356017/5384a209-7f0b-42e6-858f-70f7f724e773">

@peppy If you could make this change in the official build it would be cool 🙏🏼

**Reply by peppy** [reply]
Thanks for looking into this. I've added it in https://github.com/ppy/osu/pull/24612.

**Reply by frank89722** [reply]
@peppy The `info.plist` in v2023.924.1 of osu.app still ships with the old one with no `LSApplicationCategoryType` configured, last edited on August 23, 2021

**Reply by peppy** [reply]
I've made an additional change to make this work next release (see https://github.com/ppy/osu-deploy/commit/9d54a621f864fa6eb218286abff839fb8fde5f1f). Turns out I missed the macOS plist and only applied to iOS.
